pkg/sql/catalog/catprivilege/default_privilege.go

```diff
@@ -335,6 +335,7 @@ func expandPrivileges(
 		return
 	}
 	if GetRoleHasAllPrivilegesOnTargetObject(defaultPrivilegesForRole, targetObject) {
+		// TODO (rafi): Figure out before merging: do we still want this here?
```

Comment:
The theory here is since this case is only for `Owner` then we can remove it since `Owner` always has `ALL` right.

In PG, if you create a table as an owner, ALTER OWNER, remove the new owner role from the original owner, does the old owner have privs over the table?

If not I think we can remove it, we'd also want to test this.

Reply:
i did add a test for that, at the bottom of `grant_database`, `grant_schema`, and `grant_revoke_with_grant_option` logic tests. in PG, the old owner doesn't retain any privileges. those tests pass, so i'm not really sure what this line is doing. i'll remove it and see what breaks.

Reply:
ok i pushed a separate commit with this part removed, and with the corresponding logictest changes. i'm not too sure what to make of them.. do you know if they look good?